src/main.cpp

```diff
@@ -47,44 +49,48 @@ int main(int argc, char** argv)
 	SDL_Event e;
 
 	while (playing) {
-		currentTime = SDL_GetTicks();
-		elapsedTime = currentTime - oldTime;
-		oldTime = currentTime;
-
+		// EVENT HANDLING
 		while (SDL_PollEvent(&e) != 0) {
 			if (e.type == SDL_QUIT) {
 				playing = false;
 			} else if (e.type == SDL_MOUSEBUTTONDOWN) {
 			}
 		}
 
+
 		// GAME LOGIC
 
-		debugCounter += elapsedTime;
-		if (DEBUG_OUTPUT && debugCounter > 1000) {
-			debugCounter -= 1000;
-			debugTick = true;
-		}
 
 		// RENDERING
 		frogRenderer.clearScreen();
+		frogRenderer.renderThing(splashThing, 0, 0);// Render Splash Logo
+		frogRenderer.presentScreen();
 
-		// Render Splash Logo
-		frogRenderer.renderThing(splashThing, 0, 0);
 
-		frogRenderer.presentScreen();
+		#if DEBUG_OUTPUT // DEBUG INFO
+		debugCounter += elapsedTime;
 
-		frameTime = SDL_GetTicks() - currentTime;
+		if (DEBUG_OUTPUT && debugCounter > 1000) {
+			debugCounter -= 1000;
+			debugTick = true;
+		}
 
 		if ( debugTick ) {
 			std::cout << "Elapsed Time: " << elapsedTime << " | Frame Time: " << frameTime << std::endl;
 			debugTick = false;
 		}
+	 	#endif
+		
+
+		// FRAME HANDLING
+		currentTime = SDL_GetTicks();
+		elapsedTime = currentTime - oldTime;
+		oldTime = currentTime;
+
+		frameTime = SDL_GetTicks() - currentTime;
+
+		if (frameTime < 16) SDL_Delay(16-(SDL_GetTicks()-currentTime)); // Delay if less than 16 milliseconds
```

Comment:
I know this can be written as one line, but I find it more readable in the form it was previously.

This is my own fault, but we can also replace the duplicated logic from defining `frameTime`.
```suggestion
		if (frameTime < 16) {
			SDL_Delay(16-frameTime); // Delay if less than 16 milliseconds
		}
```